dotnet/targets/Xamarin.Shared.Sdk.props

```
@@ -8,8 +8,17 @@
 
 	<Import Project="Xamarin.Shared.Sdk.TargetFrameworkInference.props" />
 	
-	<!-- Imports the .user.env file if exists and the build is from VS -->
-	<Import Project="$(MSBuildProjectFullPath).user.env" Condition="Exists('$(MSBuildProjectFullPath).user.env') And '$(BuildingInsideVisualStudio)' == 'true'" />
+	<PropertyGroup>
+		<!-- e.g: C:\Users\user1\app1\app1\obj\ -->
+		<_MobilePropsDir>$([System.IO.Path]::GetDirectoryName($(MSBuildProjectFullPath)))\$(BaseIntermediateOutputPath)</_MobilePropsDir>
```

NBD, but there's `MSBuildProjectDirectory`, which is defined to be "The absolute path of the directory where the project file is located":

```suggestion
		<_MobilePropsDir>$(MSBuildProjectDirectory)\$(BaseIntermediateOutputPath)</_MobilePropsDir>
```

you can also use Path.Combine:


```suggestion
		<_MobilePropsDir>$([System.IO.Path]::Combine($(MSBuildProjectDirectory),$(BaseIntermediateOutputPath)))</_MobilePropsDir>
```

which solves the probing, because it will return the second path if the second path is an absolute path:

```

csharp> using System.IO;
csharp> Path.Combine ("/a", "/b")                                                                                                                                                                                                                                                                                                                                           "/b"
```

which means you wouldn't need the next line.

---

Path.Combine worked and it's cleaner than my suggestion so I went with it. PR updated